index.js

```diff
@@ -1,109 +1,145 @@
 #!/usr/bin/env node
 
-import fs from 'fs';
-import path from 'path';
+import 'babel-polyfill';
+import program from 'commander';
+import { stat, readFile, writeFile, unlink } from 'fs-extra';
 import { compileToString } from 'node-elm-compiler';
+import path from 'path';
 import sh from 'shelljs';
 
-const filename = (process.argv[2] || '').trim();
+(async () => {
```

Comment:
I put all the app's logic into a single async function, which make the data flow much easier to read. With sub-functions (which made sense before switching to async/await) the order of steps becomes a bit less obvious and there's either a need for global variables or for passing misc arguments back and forth.